### PR TITLE
ARROW-1243: [JAVA] update all libs to latest versions

### PR DESCRIPTION
**Pull request by mattdarwin**
NB this commit excludes Jackson and logback upgrades, since they are dealt with in 871 and 872

### CODE REVIEW

**Comment by jacques-n** [comment]
Most of this looks good. Let's evaluate upgrading Guava separately (not sure what it buys us and they are horrible at maintaining compatibility between versions). Can you back that out?

**Comment by mattdarwin** [comment]
guava upgrade now backed out as requested

**Comment by mattdarwin** [comment]
The following table shows every dependency in the tree (including all child dependencies).  For each, it shows the current version and date of that version, together with the date and version number of the latest version, and the number of releases between our version and the latest version.


ch.qos.logback | logback-classic | 1.0.13 | 10-May-13 |   | 1.2.3 | 31-Mar-17 | 16
-- | -- | -- | -- | -- | -- | -- | --
ch.qos.logback | logback-core | 1.0.13 | 10-May-13 | logback-classic | 1.2.3 | 31-Mar-17 | 16
com.carrotsearch | hppc | 0.7.1 | 07-May-15 |   | 0.7.2 | 25-Oct-16 | 1
com.fasterxml.jackson.core | jackson-annotations | 2.7.1 | 02-Feb-16 |   | 2.9.0 | 30-Jul-17 | 25
com.fasterxml.jackson.core | jackson-core | 2.7.1 | 02-Feb-16 | jackson-databind | 2.9.0 | 30-Jul-17 | 25
com.fasterxml.jackson.core | jackson-databind | 2.7.1 | 02-Feb-16 |   | 2.9.0 | 30-Jul-17 | 29
com.google.code.findbugs | jsr305 | 3.0.1 | 09-Oct-15 |   | 3.0.2 | 31-Mar-17 | 1
com.google.guava | guava | 18 | 25-Aug-14 |   | 22.0 | 22-May-17 | 5
com.vlkan | flatbuffers | 1.2.0-3f79e055 | 09-Nov-15 |   | 1.2.0-3f79e055 | 09-Nov-15 | 0
commons-cli | commons-cli | 1.2 | 19-Mar-09 |   | 1.4 | 09-Mar-17 | 3
commons-codec | commons-codec | 1.10 | 06-Nov-14 |   | 1.10 | 06-Nov-14 | 0
io.netty | netty-buffer | 4.0.41.Final | 29-Aug-16 | netty-handler | 4.1.13.Final | 06-Jul-17 | 16
io.netty | netty-codec | 4.0.41.Final | 29-Aug-16 | netty-handler | 4.1.13.Final | 06-Jul-17 | 16
io.netty | netty-common | 4.0.41.Final | 29-Aug-16 | netty-buffer | 4.1.13.Final | 06-Jul-17 | 16
io.netty | netty-handler | 4.0.41.Final | 29-Aug-16 |   | 4.1.13.Final | 06-Jul-17 | 16
io.netty | netty-transport | 4.0.41.Final | 29-Aug-16 | netty-handler | 4.1.13.Final | 06-Jul-17 | 16
joda-time | joda-time | 2.9 | 24-Oct-15 |   | 2.9.9 | 23-Mar-17 | 9
org.apache.commons | commons-lang3 | 3.4 | 03-Apr-15 |   | 3.6 | 09-Jun-17 | 2
org.slf4j | slf4j-api | 1.7.6 | 05-Feb-14 |   | 1.7.25 | 16-Mar-17 | 18

File: java/pom.xml [hunk 1]
```diff
@@ -526,7 +526,7 @@
     <dependency>
       <groupId>de.huxhorn.lilith</groupId>
       <artifactId>de.huxhorn.lilith.logback.appender.multiplex-classic</artifactId>
-      <version>0.9.44</version>
+      <version>8.1.1</version>
```
Comment (jacques-n) [inline java/pom.xml]: does this support java 7?

**Review by jacques-n — COMMENTED** [review]
One concern: doesn't lilith v8.x support java 7? Don't want to accidentally remove java 7 support. Otherwise, looks good

**Comment by wesm** [comment]
@mattdarwin as soon as we rebase and address the java7 question I can merge this

**Review by jacques-n — APPROVED** [review]
Lgtm

**Comment by mattdarwin** [comment]
Well spotted - Lilith 8 does indeed remove support for Java 7.  I've rolled that change back.

**Comment by wesm** [comment]
@jacques-n @siddharthteotia there were some bugs in these dependency upgrade patches fixed in  https://github.com/apache/arrow/pull/957 and https://github.com/apache/arrow/pull/960. Could someone comment whether the 0.6.0 RC is still good or whether we need to make a new RC?

**Pull request closed without merging**